lib/types/index.d.ts

```diff
@@ -599,6 +599,29 @@ export namespace Rule {
 		create(context: RuleContext): NodeListener;
 	}
 
+	type JSRuleDefinitionTypeOptions = {
+		RuleOptions: unknown[];
+		MessageIds: string;
+		ExtRuleDocs: Record<string, unknown>;
+	};
```

Comment:
We keep defining this structure in every plugin. Maybe we should make a generic `RuleDefinitionTypeOptions` in `@eslint/core` instead of copying this everywhere? @fasttime what do you think?

Reply:
We are already exporting an interface with that name (https://github.com/eslint/rewrite/blob/core-v0.13.0/packages/core/src/types.ts#L545-L556), but we could export a new type just for the options that should be customizable for a language-specific `RuleDefinition` generic type.